tuf/repository/_repository.py

```diff
@@ -188,6 +188,12 @@ def do_snapshot(
         update_version = force
         removed: Dict[str, MetaFile] = {}
 
+        root = self.root()
+        snapshot = self.snapshot()
+
+        if not root.verify_signature(snapshot):
```

Comment:
I think this should be something like:
```
snapshot_md = self.open(Snapshot.type)
root = self.root()
if not root.verify_delegate(Snapshot.type, snapshot_md.signed_bytes(), snapshot_md.signatures):
```
`self.snapshot()` returns the Snapshot object but we need the containing Metadata object (which includes the Snapshot object and the signatures).

It's really unfortunate if our linters are unable to see the issues here... we will definitely need to solve the testing situation for this code

Reply:
Oh yes, I verified snapshot and metadata directly, Got it, let me change it. Thanks for review.

Reply:
We have to use try and except block, as there is no return value so Lint is not happy.

Also, using UnsignedMetadataError and changing `bytes()` to `bytes` as lint taking it as function call.